<BlockSwitcher>

<Block name="Async/Await">

```swift
func signOutLocally() async {
    let result = await Amplify.Auth.signOut()
    guard let signOutResult = result as? AWSCognitoSignOutResult
    else {
        print("Signout failed")
        return
    }

    print("Local signout successful: \(signOutResult.signedOutLocally)")
    switch signOutResult {
    case .complete:
        print("SignOut completed")
    case .failed(let error):
        print("SignOut failed with \(error)")
    case let .partial(revokeTokenError, globalSignOutError, hostedUIError):
        print(
            """
            SignOut is partial.
            RevokeTokenError: \(String(describing: revokeTokenError))
            GlobalSignOutError: \(String(describing: globalSignOutError))
            HostedUIError: \(String(describing: hostedUIError))
            """
        )
    }
}
```

</Block>

<Block name="Combine">

```swift
func signOutLocally() -> AnyCancellable {
    Amplify.Publisher.create {
        await Amplify.Auth.signOut()
    }.sink(receiveValue: { result in
        guard let signOutResult = result as? AWSCognitoSignOutResult
        else {
            print("Signout failed")
            return
        }
        print("Local signout successful: \(signOutResult.signedOutLocally)")
        switch signOutResult {
        case .complete:
            print("SignOut completed")
        case .failed(let error):
            print("SignOut failed with \(error)")
        case let .partial(revokeTokenError, globalSignOutError, hostedUIError):
            print(
                """
                SignOut is partial.
                RevokeTokenError: \(String(describing: revokeTokenError))
                GlobalSignOutError: \(String(describing: globalSignOutError))
                HostedUIError: \(String(describing: hostedUIError))
                """
            )
        }
    })
}
```

</Block>

</BlockSwitcher>
